### PR TITLE
Validations

### PR DESCRIPTION
**Pull request by pntgupta**

### CODE REVIEW

File: client/app/shared/base.component.ts [hunk 1]
```diff
@@ -149,4 +161,27 @@ export class BaseComponent implements OnInit, OnDestroy {
         }
         return this.modelRef["disabled"];
     }
-}
\ No newline at end of file
+
+    addValidations() {
+        //Comp dynamically created (unlike label component)
+        if (this.viewNumber) {
```
Comment (aditi-kapur) [inline client/app/shared/base.component.ts]: Please add a comment why check for viewNumber is made

File: client/app/shared/tab-page/tab-page.component.ts [hunk 2]
```diff
@@ -16,20 +16,20 @@ export class TabPageComponent extends ContainerComponent {
     this.factoryRef = injector.get(InputFactoryService);
   }
 
-  AddChildElements(factoryRef, containerRef, itemArray) {
+  AddChildElements(factoryRef, containerRef, itemArray, viewNumber) {
     if (this.modelRef) {
       for (let item of itemArray) {
         let childModelRef = this.modelRef[item.relVal];
-        factoryRef.createComp(containerRef, item, childModelRef);
+        factoryRef.createComp(containerRef, item, childModelRef, viewNumber);
```
Comment (aditi-kapur) [inline client/app/shared/tab-page/tab-page.component.ts]: viewNumber is added as a third param in createComp function in file: input-factory.service.ts

Comment (pntgupta) [reply client/app/shared/tab-page/tab-page.component.ts]: Error fixed.

File: client/app/shared/base.component.ts [hunk 3]
```diff
@@ -13,22 +14,32 @@ export class BaseComponent implements OnInit, OnDestroy {
     descriptionConfig: itemSchema = new itemSchema();
     eventSrvc: Object;
     subscriptions: Array<Object>;
+    viewNumber: number;
```
Comment (aditi-kapur) [inline client/app/shared/base.component.ts]: Remove this variable. Not used now.

**Review by dhirendra-singh81 — APPROVED** [review]

**Pull request merged**